src/sphinx_book_theme/__init__.py

```diff
@@ -81,20 +85,37 @@ def hash_assets_for_files(assets: list, theme_static: Path, context):
     """
     for asset in assets:
         # CSS assets are stored in css_files, JS assets in script_files
-        asset_type = "css_files" if asset.endswith(".css") else "script_files"
-        if asset_type in context:
-            # Define paths to the original asset file, and its linked file in Sphinx
-            asset_sphinx_link = f"_static/{asset}"
-            asset_source_path = theme_static / asset
-            if not asset_source_path.exists():
-                SPHINX_LOGGER.warn(
-                    f"Asset {asset_source_path} does not exist, not linking."
-                )
-            # Find this asset in context, and update it to include the digest
-            if asset_sphinx_link in context[asset_type]:
-                hash = _gen_hash(asset_source_path)
-                ix = context[asset_type].index(asset_sphinx_link)
-                context[asset_type][ix] = asset_sphinx_link + "?digest=" + hash
+        if asset.endswith(".css"):
+            asset_context = context.get("css_files", [])
+        elif asset.endswith(".js"):
+            asset_context = context.get("js_files", [])
+        else:
+            SPHINX_LOGGER.warn(f"Unrecognised asset type, not hashing.")
+            continue
+
+        # Define paths to the original asset file, and its linked file in Sphinx
+        asset_sphinx_link = f"_static/{asset}"
+        asset_source_path = theme_static / asset
+        if not asset_source_path.exists():
+            SPHINX_LOGGER.warn(
+                f"Asset {asset_source_path} does not exist, not linking."
+            )
+            continue
+
+        # Find this asset in context, and update it to include the digest
+        try:
+            ix = asset_context.index(asset)
+        except ValueError:
+            continue
+
+        asset_hash = _gen_hash(asset_source_path)
+
+        asset_obj = asset_context[ix]
+        asset_context[ix] = type(asset_obj)(
```

Comment:
Is it not possible to update the filename in-place for these assets? We have to re-create it?